src/Server.cpp

```diff
@@ -35,6 +38,36 @@ RouteCfg* find_route(const HttpRequest& req, std::vector<RouteCfg>& routes)
     return route_match.second;
 }
 
+/*
+*   @check_request_method:
+*    Checks if the requested method is accepted by the route
+*/
+bool    check_request_method(RouteCfg* route, const std::string method) {
+    for (size_t i = 0; i < route->accepted_methods.size(); i++)
+        if (method == route->accepted_methods[i])
+            return true;
+    return false;
+}
+
+/*
+*   @index_path:
+*       If auto_index is true, returns user request
+*       If not, return predefined index
+*
+*/
+std::string	index_path(const HttpRequest& req, RouteCfg* route)
+{
+    std::string	request = "." + route->root + route->route_path;
+
+    if ( route->auto_index ) {
+        if ( req.target() == "/" ) return request;
+        if ( route->route_path == "/") return request + req.target().substr(1, req.target().size() - 1);
+
+        return "." + route->root + req.target();
```

Comment:
I am having trouble understanding this, please comment with a thorough explaination